core/network/corda-interop-app/interop-workflows/src/main/kotlin/com/weaver/corda/app/interop/flows/AssetTransferFlows.kt

```diff
@@ -474,7 +474,7 @@ object ReclaimPledgedAsset {
 
             val viewData = subFlow(GetExternalStateByLinearId(claimStatusLinearId))
             val externalStateView = ViewDataOuterClass.ViewData.parseFrom(viewData)
-            val payloadDecoded = Base64.getDecoder().decode(externalStateView.payload.toByteArray())
+            val payloadDecoded = Base64.getDecoder().decode(externalStateView.notarizedPayloadsList[0].payload.toByteArray())
```

Comment:
Instead of taking the first array element, can you insert a function call here? In the function (taking payloads, private key, and verification policy as parameters), ensure that all payloads match each other and return that or an error if there is a mismatch. Add a TODO saying that we need to add more logic to match encrypted payloads and also determine whether there are enough matching payloads to pass the verification policy.

I think we will need to call this function in `WriteExternalState.kt` too.